upload-api/billing.js

```
@@ -0,0 +1,89 @@
+    },
+
+    async setPlan(customerDID, plan) {
+      /** @type {import('stripe').Stripe.SubscriptionItem[] | undefined} */
+      let subscriptionItems
+      /** @type {string | undefined} */
+      let priceID
+      try {
+        const prices = await stripe.prices.list({ lookup_keys: [plan] })
+        priceID = prices.data.find(price => price.lookup_key === plan)?.id
+        if (!priceID) return (
+          { error: new InvalidSubscriptionState(`could not find Stripe price with lookup_key ${plan} - cannot set plan`) }
+        )
+
+        const email = toEmail(/** @type {import('@web3-storage/did-mailto').DidMailto} */(customerDID))
+        const customers = await stripe.customers.list({ email, expand: ['data.subscriptions'] })
+        if (customers.data.length != 1) return (
+          { error: new InvalidSubscriptionState(`found ${customers.data.length} Stripe customer(s) with email ${email} - cannot set plan`) }
+        )
+
+        const customer = customers.data[0]
+        const subscriptions = customer.subscriptions?.data
+        if (subscriptions?.length != 1) return (
+          { error: new InvalidSubscriptionState(`found ${subscriptions?.length} Stripe subscriptions(s) for customer with email ${email} - cannot set plan`) }
+        )
+
+        subscriptionItems = customer.subscriptions?.data[0].items.data
+        if (subscriptionItems?.length != 1) return (
+          { error: new InvalidSubscriptionState(`found ${subscriptionItems?.length} Stripe subscriptions item(s) for customer with email ${email} - cannot set plan`) }
+        )
+      } catch (/** @type {any} */ err) {
+        return { error: new InvalidSubscriptionState(err.message, { cause: err }) }
+      }
+
+      try {
+        await stripe.subscriptionItems.update(subscriptionItems[0].id, { price: priceID })
+
+        return { ok: {} }
+      } catch (/** @type {any} */ err) {
+        return { error: new BillingProviderUpdateError(err.message, { cause: err }) }
```

I don't know if it's worth wrapping a completely unexpected error...personally I'd let this throw.